src/components/Popover/index.tsx

```diff
@@ -253,7 +253,11 @@ const Popover = React.forwardRef<HTMLElement, PopoverProps>(
       >
         <Popper
           innerRef={(node) => {
-            (refContent as MutableRefObject<HTMLElement>).current = node;
+            if (typeof refContent === 'function') {
+              refContent(node);
+            } else {
+              (refContent as MutableRefObject<HTMLElement>).current = node;
+            }
```

Comment:
Nice, is this a type guard? I would love a walkthrough of how you came to decide that this needed to be done.

Reply:
the thing here is `refContent` can be either a function or a mutable ref, if it's a function it works differently, that's why there's the check for it, if it's not a function then we're sure it's a mutable ref, but typescript still doesn't like it so we have to assert it.

`forwardedRef: ((instance: HTMLElement | null) => void) | React.MutableRefObject<HTMLElement | null> | null`
forwardedRef comes from `React.forwardRef` that is wrapping the component.

Reply:
hmmm. Is there any possible way to not have to use `as` here?

Reply:
i'm not sure, i can't think of one, i don't know if there's something more advanced in typescript that enables this for us, i also tried `typeof` or `instanceof` and it didn't work out, but what i do know is we need it to be a `MutableRefObject` so it allows us to change the value of `current`

Reply:
Ok, that's fine for now.